spec/services/generate_test_data_spec.rb

```diff
@@ -22,5 +22,13 @@
         expect(application_choice.provider.code).to eq 'DEF'
       end
     end
+
+    it 'works when duplicate course codes are generated' do
+      provider = create(:provider, code: 'DEF')
+      Faker::Config.random = Random.new(42)
+      GenerateTestData.new(1, provider).generate
+      Faker::Config.random = Random.new(42)
+      GenerateTestData.new(1, provider).generate
+    end
```

Comment:
where is the expectation?

Reply:
It throws an exception without the fix but you are right, should have an expectation. I guess I could just add an expectation to assert no exception, but I think asserting that an application form is created on each call to `generate` is best.